src/fn/upload-docs.js

```diff
@@ -0,0 +1,69 @@
+const fs = require('../lib/sync-fs');
+const info = require('../lib/log').info;
+const pouch = require('../lib/db');
+const skipFn = require('../lib/skip-fn');
+const trace = require('../lib/log').trace;
+const warn = require('../lib/log').warn;
+
+const BATCH_SIZE = 2; // increase to 100
+
+module.exports = (projectDir, couchUrl) => {
+  if(!couchUrl) return skipFn('no couch URL set');
+
+  const docDir = `${projectDir}/json_docs`;
+  const db = pouch(couchUrl);
+
+  if(!fs.exists(docDir)) {
+    warn(`No docs directory found at ${docDir}.`);
+    return Promise.resolve();
+  }
+
+  let docFiles = fs.recurseFiles(docDir)
+    .filter(name => name.endsWith('.doc.json'));
+
+  docFiles
+    .forEach(validateJson);
+
+  const sets = [];
+  while(docFiles.length) {
+    sets.push(docFiles.slice(0, BATCH_SIZE));
+    docFiles = docFiles.slice(BATCH_SIZE);
+  }
+
+  info(`Uploading docs in ${sets.length} sets…`);
+
+  const results = { ok:[], failed:{} };
+
+  const process = sets.reduce((promiseChain, docSet) =>
+    promiseChain.then(() =>
+      db.bulkDocs(docSet.map(fs.readJson))
+        .then(res => {
+          trace('Uploaded', docSet);
+          res.forEach(r => {
+            if(r.error) {
+              results.failed[r.id] = r.error;
+            } else {
+              results.ok.push(r.id);
```

Comment:
It'd be good to delete, or maybe just change the extension, of successfully imported docs. This means if it fails and you rerun it it won't try and upload the same docs again and get conflicts.

Reply:
Nice idea.